source/internal/_clone.js

```diff
@@ -25,9 +25,12 @@ export default function _clone(value, refFrom, refTo, deep) {
     refFrom[idx] = value;
     refTo[idx] = copiedValue;
     for (var key in value) {
-      copiedValue[key] = deep ?
-        _clone(value[key], refFrom, refTo, true) : value[key];
+      if (value.hasOwnProperty(key)) {
+        copiedValue[key] = deep ?
+          _clone(value[key], refFrom, refTo, true) : value[key];
+      }
     }
+    copiedValue.__proto__ = value.__proto__;
```

Comment:
Took a look at MDN and it looks like using `__proto__` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/proto). Is there an alternative way to accomplish this?

Reply:
@Bradcomp 
Thanks for the advice!
I try to use the build-in function `Object.prototype.create` and `Object.prototype.getPrototypeOf` to replace this.
I fixed it in eaa0ac9, please check!